devito/tools.py

```diff
@@ -569,3 +570,20 @@ def __new__(cls, *items, **kwargs):
         obj = super(EnrichedTuple, cls).__new__(cls, items)
         obj.__dict__.update(kwargs)
         return obj
+
+
+class sparse_fd_list(list):
```

Comment:
I suggest to move this to ``finite_differences``. ``tools`` is orthogonal to FD helpers/concepts